android/app/src/main/java/org/ZingoLabs/Zingo/BackgroundSyncWorker.kt

```diff
@@ -0,0 +1,125 @@
+package org.ZingoLabs.Zingo
+
+import android.content.Context
+import androidx.work.Worker
+import androidx.work.WorkerParameters
+import android.util.Log
+import java.io.File
+import java.util.*
+import org.json.JSONObject
+import java.io.InputStream
+import java.nio.charset.StandardCharsets
+import com.facebook.react.bridge.ReactApplicationContext
+
+class BackgroundSyncWorker(context: Context, workerParams: WorkerParameters) : Worker(context, workerParams) {
+
+    override fun doWork(): Result {
+        val reactContext = ReactApplicationContext(MainApplication.getAppContext())
+        val rpcModule = RPCModule(reactContext)
+
+        Log.w("SCHEDULED_TASK_RUN", "Task running")
```

Comment:
I'd recommend switching the warning logs here to info or debug level, as most of them seem to be rather informative. But it depends on your approach.

Reply:
Thanks, good idea.